File: BankOfMyHouse.Server/BankOfMyHouse.API/Endpoints/Transactions/CreateTransaction/CreateTransactionRequestDto.cs
using BankOfMyHouse.API.Endpoints.Transactions.DTOs;
using BankOfMyHouse.Domain.BankAccounts;
using FastEndpoints;

namespace BankOfMyHouse.API.Endpoints.Transactions.CreateTransaction;

public sealed record CreateTransactionRequestDto
{
	[FromClaim("userId")]
	public int UserId { get; set; }

	public required IbanCodeDto SenderIban { get; set; }
	public required IbanCodeDto ReceiverIban { get; set; }
	public required decimal Amount { get; set; }
	public required string CurrencyCode { get; set; }
	public PaymentCategoryCode PaymentCategory { get; set; } = PaymentCategoryCode.Other;
	public string? Description { get; set; }
}

File: BankOfMyHouse.Client/src/app/transactions/models/createTransactionRequestDto.ts
export class CreateTransactionRequestDto {
    senderIban: IbanCodeDto = new IbanCodeDto('');
    receiverIban: IbanCodeDto = new IbanCodeDto('');
    amount: number = 0;
    currencyCode: string = '€';
    paymentCategory: PaymentCategoryCode = PaymentCategoryCode.Other;
    description: string = '';
}

export class IbanCodeDto {
    constructor(value: string) {
        this.value = value.trim();
    }

    value: string = '';
}

export enum PaymentCategoryCode {
    Other,
    Shopping,
    Food,
    Transport,
    Utilities,
    Entertainment,
    Healthcare,
    Education,
    Salary,
    Investment,
    Savings,
    Rent,
    Insurance
}